clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -338,6 +338,20 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
     if (type_is_instr(memref.instr.type) ||
         memref.instr.type == TRACE_TYPE_PREFETCH_INSTR ||
         memref.instr.type == TRACE_TYPE_INSTR_NO_FETCH) {
+        bool expect_encoding = TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, shard->file_type_);
+        std::unique_ptr<instr_t> cur_instr_decoded = nullptr;
+        if (expect_encoding && memref.instr.encoding_is_new) {
+            cur_instr_decoded.reset(new instr_t);
+            instr_init(GLOBAL_DCONTEXT, cur_instr_decoded.get());
+            app_pc next_pc = nullptr;
```

Comment:
The declaration of next_pc and the assignment in the next line don't need to be separate.

Reply:
Ack. This was to use the default `nullptr` for the check below but looks like `decode_from_copy` returns NULL on failure. So will combine it as suggested.